### PR TITLE
Add initial edoc config and docs/

### PR DESCRIPTION
**Pull request by yurrriq**
Generate Github Flavoured Markdown in `doc/`.

**Note**: This depends on #19, but doesn't really need to.
**Related**: #4

### CODE REVIEW

File: bin/set-edoc-hidden [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env sh
+
+[ -f "$1" ] || exit 1
+
+FILE="$1"
+HIDDEN="%% @hidden"
+
+if [ "$(head -n 1 $FILE)" != "$HIDDEN" ]; then
+    echo "0a\n$HIDDEN\n.\nw" | ed "$FILE" >/dev/null
```
Comment (yurrriq) [inline bin/set-edoc-hidden]: :laughing:

**Comment by yurrriq** [comment]
Rebased on master.

File: bin/set-edoc-hidden (hunk 1 above)
Comment (deadtrickster) [inline bin/set-edoc-hidden]: Nice to see you adopted bin folder!
What the purpose of this script? it adds `@hidden` tag to `FILE`?

**Comment by yurrriq** [comment]
As noted in #4, we can use

```erlang
{edoc_opts, [{dir, "docs"}]}.
```

to generate (probably) [hexdocs](https://hexdocs.pm)-friendly HTML docs in `docs/` instead. Maybe we could do this in separate profiles.

File: bin/set-edoc-hidden (hunk 1 above)
Comment (yurrriq) [inline bin/set-edoc-hidden]: If the first line is not `%% @hidden` then it prepends it.

**Comment by deadtrickster** [comment]
I talked to Eric (probably it was him) and he said everything with docs/index.html is fine.

File: src/model/prometheus_model.erl [hunk 2]
```diff
@@ -77,40 +78,40 @@ e_msg_Bucket(Msg, TrUserData) ->
 
 
 e_msg_Bucket(#'Bucket'{cumulative_count = F1,
-           upper_bound = F2},
-       Bin, TrUserData) ->
+		       upper_bound = F2},
```
Comment (deadtrickster) [inline src/model/prometheus_model.erl]: Tabs :-(

Comment (yurrriq) [inline src/model/prometheus_model.erl]: 3dae9f2

**Comment by yurrriq** [comment]
[HTML preview](https://htmlpreview.github.io/?https://github.com/yurrriq/prometheus.erl/blob/feature/edoc-config/docs/index.html)

**Comment by deadtrickster** [comment]
Why tabs sneaked anyway? Indenting prometheus_model manually doesn't look like a good idea because it auto-generated.

**Comment by yurrriq** [comment]
I think I manually `untabify`'d it before too. I have save hooks that re-indent, and/or automatic key chord impulses..

**Comment by yurrriq** [comment]
```fish
gsed -i 's/\t/  /g' src/model/prometheus_model.erl
```

The default macOS `sed` doesn't have `-i` IIRC. It must be GNU.

**Comment by deadtrickster** [comment]
Yea I have indent-whole-buffer (bound to F12) too and I like to smash it. 
And you are right, gpb generates it with tabs. Can you please reset it 'back' to Tabs then so we wouldn't have this huge diff in the future anymore.

**Comment by deadtrickster** [comment]
I update gpb/rebar3_gpb_plugin one more time. As it turned out elvis just skipped src/model :confused:.
Now elvis checks just prometheus_model_helpers. I as long as we add only `@hidden` I'm fine with modifications.

**Comment by deadtrickster** [comment]
Lets delete this commit https://github.com/deadtrickster/prometheus.erl/pull/20/commits/3c664ce3b8502ce5863c30ffc15f450728bd83ca and I'll merge PR.

**Comment by deadtrickster** [comment]
Thanks!

**Pull request merged**